File: src/main/java/me/nobaboy/nobaaddons/features/chatcommands/PartyCommands.java
package me.nobaboy.nobaaddons.features.chatcommands;

import me.nobaboy.nobaaddons.NobaAddons;
import me.nobaboy.nobaaddons.features.chatcommands.impl.party.*;
import me.nobaboy.nobaaddons.features.chatcommands.impl.shared.HelpCommand;
import net.minecraft.util.StringUtils;
import net.minecraftforge.client.event.ClientChatReceivedEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

import java.util.Optional;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class PartyCommands extends ChatCommandManager {
    // .*? is for the player symbols added by mods
    private static final Pattern chatPattern = Pattern.compile("^Party > .*?(?:\\[[A-Z+]+] )?(?<username>[A-z0-9_]+).*?: !(?<command>[A-z0-9_]+) ?(?<argument>[A-z0-9_]+)?");

    {
        register(new HelpCommand(this, "pc", () -> NobaAddons.config.partyHelpCommand));
        register(new TransferCommand());
        register(new AllInviteCommand());
        register(new WarpCommand());
        register(new CancelCommand());
        register(new CoordsCommand());
    }

    @Override
    protected Optional<Matcher> matchMessage(String message) {
        Matcher match = chatPattern.matcher(message);
        if(match.find()) {
            return Optional.of(match);
        }
        return Optional.empty();
    }

    @SubscribeEvent
    public void onChatReceived(final ClientChatReceivedEvent event) {
        if(!NobaAddons.config.partyCommands) return;
        String receivedMessage = StringUtils.stripControlCodes(event.message.getUnformattedText());

        processMessage(receivedMessage);
    }
}

File: src/main/kotlin/me/nobaboy/nobaaddons/features/chatcommands/impl/PartyCommands.kt
package me.nobaboy.nobaaddons.features.chatcommands.impl

import me.nobaboy.nobaaddons.NobaAddons
import me.nobaboy.nobaaddons.features.chatcommands.ChatCommandManager
import me.nobaboy.nobaaddons.features.chatcommands.impl.party.*
import me.nobaboy.nobaaddons.features.chatcommands.impl.shared.HelpCommand
import me.nobaboy.nobaaddons.util.StringUtils.cleanString
import me.nobaboy.nobaaddons.util.RegexUtils.matchMatcher
import net.minecraftforge.client.event.ClientChatReceivedEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import java.util.regex.Matcher
import java.util.regex.Pattern

class PartyCommands : ChatCommandManager() {
    private val config get() = NobaAddons.config.chatCommands.partyCommands
    private val chatPattern: Pattern =
        Pattern.compile("^Party > .*?(?:\\[[A-Z+]+] )?(?<username>[A-z0-9_]+).*?: [!?.](?<command>[A-z0-9_]+) ?(?<argument>[A-z0-9_]+)?")

    init {
        register(HelpCommand(this, "pc", config::help))
        register(TransferCommand())
        register(AllInviteCommand())
        register(WarpCommand())
        register(CancelCommand())
        register(CoordsCommand())
    }

    override fun matchMessage(message: String): Matcher? {
        chatPattern.matchMatcher(message) {
            return this
        }
        return null
    }

    @SubscribeEvent
    fun onChatReceived(event: ClientChatReceivedEvent) {
        if (!isEnabled()) return

        val receivedMessage = event.message.unformattedText.cleanString()
        processMessage(receivedMessage)
    }

    fun isEnabled() = config.enabled
}